pkg/restore/restore.go

```diff
@@ -732,6 +742,15 @@ func (ctx *restoreContext) restoreResource(resource, targetNamespace, originalNa
 
 	groupResource := schema.ParseGroupResource(resource)
 
+	// Modify `resource` so that it has the priority version to restore in its
+	// path. For example, for group resource "horizontalpodautoscalers.autoscaling",
+	// "/v2beta1" will be appended to the end. Different versions would only
+	// have been stored if the APIGroupVersionsFeatureFlag was enabled during backup.
+	cgv, ok := ctx.chosenGrpVersToRestore[groupResource.String()]
```

Comment:
I would add more comments here. the `chosenGrpVersToRestore` would exist if meetsAPIGVRestoreReqs as well. not only if the backup was taken with featureflag.

Reply:
I will add a sentence to the comment:
```
// The chosenGrpVersToRestore map would only be populated if APIGroupVersionsFeatureFlag
// was enabled for restore and the minimum required backup format version has been met.
```